src/datanode/src/error.rs

```diff
@@ -359,6 +359,13 @@ pub enum Error {
         #[snafu(implicit)]
         location: Location,
     },
+
+    #[snafu(display("Failed to serialize options to TOML"))]
+    TomlFormat {
+        #[snafu(implicit)]
+        location: Location,
+        source: Box<common_config::error::Error>,
```

Comment:
Try https://docs.rs/snafu/latest/snafu/derive.Snafu.html#transforming-the-source